subprojects/base-services/src/main/java/org/gradle/internal/operations/DefaultBuildOperationQueue.java

```diff
@@ -164,11 +165,11 @@ public void run() {
         }
 
         private void runBuildOperation() {
-            BuildOperationWorkerRegistry.Completion workerLease = owner.operationStart();
+            WorkerLeaseRegistry.WorkerLeaseCompletion workerLease = owner.startChild();
```

Comment:
I wonder if we can replace this pattern with the more generic "release a lock" pattern, i.e.

```
coordinator.withoutLock(new WorkerLease(), new Action() {
  worker.execute(operation)
})
```

Then we'd no longer have to pass around lock instances, but could just say "I release one worker lease" as shown above.